Preserve non-200 status when no content will be sent

PradnyaBaviskar opened the pull request:
to / @bajtos -
In reference to
1) Strong-remoting issue - https://github.com/strongloop/strong-remoting/issues/175 , and
2) LB issue - https://github.com/strongloop/loopback/issues/416

I've made changes to retain non-200 status, when no content is sent.

Can you please review?

bajtos commented:
@PradnyaBaviskar please add a unit-test to verify the change works as intended.

bajtos commented:
This allows developers to write a remote method that calls the final callback and at the same time sets the response to 304 redirect, either via accepting the `req` object as an argument or via a remote hook.

```js
UserModel.afterRemote('confirm', function(ctx, inst, next) {
  if (ctx.req.query.redirectUrl !== undefined) {
    ctx.res.location(redirectUrl);
    ctx.res.status(302);
  }
  next();
});
```

@ritch you asked me to let you review all strong-remoting changes, could you please take a look?

ritch commented:
I would also add if status code is not defined set it to 204.

Feel free to merge as is though.

PradnyaBaviskar commented:
@bajtos ,
1) I've added test case to verify that non-200 status is preserved.
2) Also modified the code to set response status to 200, if it is undefined.

Can you please review?

bajtos commented on test/rest.test.js:
```diff
@@ -722,6 +722,25 @@ describe('strong-remoting-rest', function() {
+         },
+        {
+          accepts: [
+            { arg: 'ctx', type: 'object', http: {source: 'context' } }
```
FYI: you can also use `{arg: 'res', type: 'object', http: { source: 'res' } }` and then access the response directly.

bajtos commented:
Landed, thank you!
